src/app/calculator/pumps/achievable-efficiency/achievable-efficiency.component.ts

```diff
@@ -71,44 +73,82 @@ export class AchievableEfficiencyComponent implements OnInit {
     if (!this.psat) {
       this.achievableEfficiencyService.flowRate = this.efficiencyForm.controls.flowRate.value;
       this.achievableEfficiencyService.pumpType = this.efficiencyForm.controls.pumpType.value;
+      this.pumpEfficiencyInputs = this.achievableEfficiencyService.getObjectFromForm(this.efficiencyForm);
+      this.achievableEfficiencyService.pumpEfficiencyInputs = this.pumpEfficiencyInputs;
     }
   }
 
   initForm() {
     if (!this.psat) {
       //patch default/starter values for stand alone calculator
       if (this.achievableEfficiencyService.flowRate && this.achievableEfficiencyService.pumpType) {
-        this.efficiencyForm = this.achievableEfficiencyService.getForm(this.achievableEfficiencyService.pumpType, this.achievableEfficiencyService.flowRate);
+        this.efficiencyForm = this.achievableEfficiencyService.getForm(this.achievableEfficiencyService.pumpEfficiencyInputs);
       }
       else {
         let tmpFlowRate: number = 2000;
         if (this.settings.flowMeasurement !== 'gpm') {
           tmpFlowRate = this.convertUnitsService.value(tmpFlowRate).from('gpm').to(this.settings.flowMeasurement);
           tmpFlowRate = this.psatService.roundVal(tmpFlowRate, 2);
         }
-        this.efficiencyForm = this.achievableEfficiencyService.getForm(6, tmpFlowRate);
+        let inputs: PumpEfficiencyInputs = {
```

Comment:
Looking at these default constants, There may be a need to check settings unitsOfMeasure here.